ansible/roles/docker/tasks/main.yml

```diff
@@ -5,6 +5,9 @@
 - include: apt-docker-install.yml
   when: ansible_distribution == "Ubuntu"
 
+- include: use-docker-repos.yml
+  when: docker_provider == "docker"
```

Comment:
I would prefer ``docker_source_type == custom_repository``. As currently, there are ``etcd_source_type``, ``flannel_source_type`` and upcoming ``kube_source_type``, it make sense to continue in the naming schema. At the same time, there are various ways how to install a component. Such as:

* package manager
* github release
* local build
* distribution rpm
* custom repository

I have already a usecase for choosing the custom repository. So this would work for both of us. At the same time it would be beneficial to make the repository file configurable. E.g.

```ini
[dockerrepo]
name=Docker Repository
baseurl={{ repository_baseurl }}
enabled=1
gpgcheck={% if repository_gpgkey == "" %}0{% else %}1{% endif %}
gpgkey={{ repository_gpgkey }}
```

```yaml
# Docker custom repository
repository_baseurl: https://yum.dockerproject.org/repo/main/centos/$releasever/
repository_gpgkey: https://yum.dockerproject.org/gpg
```